### PR TITLE
Enable huge page for posix platforms

### PR DESCRIPTION
**Pull request by wenyongh**
This patch enables huge page support for posix platforms for performance
reason, if the request size to mmap is larger than 2MB, then we use madvise
to set some pages to huge page.
And add macro control to enable tracing the mmap/munmap.

### CODE REVIEW

File: core/shared/platform/common/posix/posix_memmap.c [hunk 1]
```diff
@@ -5,24 +5,55 @@
...
     int map_flags = MAP_ANONYMOUS | MAP_PRIVATE;
     uint64 request_size, page_size;
-    uint8 *addr;
+    uint8 *addr = NULL;
     uint32 i;
 
     page_size = (uint64)getpagesize();
     request_size = (size + page_size - 1) & ~(page_size - 1);
 
+#ifndef __APPLE__ /* huge page isn't supported on MacOS */
```
Comment (no1wudi) [inline core/shared/platform/common/posix/posix_memmap.c:45]:
```suggestion
#if  !defined(__APPLE__) && !defined(__NuttX__) /* huge page isn't supported on MacOS and NuttX */
```

File: core/shared/platform/common/posix/posix_memmap.c [hunk 2]
```diff
@@ -82,23 +113,89 @@ os_mmap(void *hint, size_t size, int prot, int flags)
...
+    }
+
+#if BH_ENABLE_TRACE_MMAP != 0
+    total_size_mmapped += request_size;
+    os_printf("mmap return: %p with size: %zu, total_size_mmapped: %zu, "
+              "total_size_munmapped: %zu\n",
+              addr, request_size, total_size_mmapped, total_size_munmapped);
+#endif
+
+#ifndef __APPLE__ /* huge page isn't supported on MacOS */
```
Comment (no1wudi) [inline core/shared/platform/common/posix/posix_memmap.c:147]:
```suggestion
#if  !defined(__APPLE__) && !defined(__NuttX__) /* huge page isn't supported on MacOS and NuttX */
```

File: core/shared/platform/common/posix/posix_memmap.c [hunk 3]
```diff
@@ -82,23 +113,89 @@ os_mmap(void *hint, size_t size, int prot, int flags)
...
+            if (ret) {
+#if BH_ENABLE_TRACE_MMAP != 0
+                os_printf(
+                    "warning: madvise(%p, %lu) huge page failed, return %d\n",
+                    (void *)huge_start, huge_end - huge_start, ret);
+#endif
+            }
+        }
+    }
+#endif /* end of __APPLE__ */
```
Comment (no1wudi) [inline core/shared/platform/common/posix/posix_memmap.c:198]:
```suggestion
#endif /* end of __APPLE__ and __NuttX__ */
```

**Review by no1wudi — CHANGES_REQUESTED** [review]

**Review by no1wudi — APPROVED** [review]
LGTM

File: core/shared/platform/common/posix/posix_memmap.c (hunk 1 above)
Comment (wenyongh) [reply core/shared/platform/common/posix/posix_memmap.c]: Done

File: core/shared/platform/common/posix/posix_memmap.c (hunk 3 above)
Comment (wenyongh) [reply core/shared/platform/common/posix/posix_memmap.c]: Done

**Pull request merged**